tests/tests/test_docker_compose.py

```diff
@@ -0,0 +1,218 @@
+# Copyright 2025 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import os
+import subprocess
+import tempfile
+from contextlib import contextmanager
+from dataclasses import dataclass
+
+from ..common_setup import standard_setup_virtualization
+from .common_update import common_update_procedure
+from ..MenderAPI import DeviceAuthV2, Deployments
+from .mendertesting import MenderTesting
+
+
+@dataclass
+class DockerService:
+    name: str
+    image: str
+    base_image: str = "busybox:latest"
+    build_image: bool = True
+
+
+@contextmanager
+def create_test_manifest(services):
+    """Create Docker images and compose manifest for testing."""
+    with tempfile.TemporaryDirectory() as temp_dir:
+        manifests_dir = os.path.join(temp_dir, "manifests")
+        images_dir = os.path.join(temp_dir, "images")
+        os.makedirs(manifests_dir)
+        os.makedirs(images_dir)
+
+        # Build and save images for each service that needs it
+        for service in services:
+            if service.build_image:
+                dockerfile = os.path.join(temp_dir, f"Dockerfile.{service.name}")
+                with open(dockerfile, "w") as f:
+                    f.write(f'FROM {service.base_image}\nCMD ["sleep", "infinity"]\n')
+                subprocess.check_call(
+                    ["docker", "build", "-t", service.image, "-f", dockerfile, temp_dir]
+                )
+                subprocess.check_call(
+                    [
+                        "docker",
+                        "save",
+                        "-o",
+                        os.path.join(images_dir, f"{service.image}.tar"),
+                        service.image,
+                    ]
+                )
+
+        with open(os.path.join(manifests_dir, "docker-compose.yml"), "w") as f:
+            f.write("services:\n")
+            for service in services:
+                f.write(f"  {service.name}:\n")
+                f.write(f"    image: {service.image}\n")
+                f.write(f"    network_mode: bridge\n")
+
+        yield manifests_dir, images_dir
+
+
+def make_docker_compose_artifact(manifests_dir, project_name, images_dir=None):
```

Comment:
Does it work at all with `images_dir=None`? Please remove the default value, this is a mandatory argument.